core/domain/translation_domain.py

```diff
@@ -18,9 +18,285 @@
+
+    HTML = 'html'
+    UNICODE_STRING = 'unicode'
+    SET_OF_NORMALIZED_STRING = 'set_of_normalized_string'
+    SET_OF_UNICODE_STRING = 'set_of_unicode_string'
+
+
+class TranslatableContentDict(TypedDict):
+    """Dictionary representing TranslatableContent object."""
+
+    content_id: str
+    content_value: feconf.ContentValueType
+    content_type: TranslatableContentFormat
+
+
+class TranslatableContent:
+    """TranslatableContent represents a content of a translatable object which
+    can be translated into multiple languages.
+
+    Args:
+        content_id: str. The id of the corresponding translatable content value.
+        content_value: ContentValueType. The content value which can be
+            translated.
+        content_type: TranslatableContentFormat. The type of the
+            corresponding content value.
+    """
+
+    def __init__(
+        self,
+        content_id: str,
+        content_value: feconf.ContentValueType,
+        content_type: TranslatableContentFormat
+    ) -> None:
+        self.content_id = content_id
+        self.content_value = content_value
+        self.content_type = content_type
+
+    @classmethod
+    def from_dict(
+        cls,
+        translatable_content_dict: TranslatableContentDict
+    ) -> TranslatableContent:
+        """Returns a TranslatableContent object from its dict representation.
+
+        Args:
+            translatable_content_dict: dict. Dict representation of
+                TranslatableContent object.
+
+        Returns:
+            TranslatableContent. An instance of TranslatableContent class.
+        """
+        return cls(
+            translatable_content_dict['content_id'],
+            translatable_content_dict['content_value'],
+            translatable_content_dict['content_type'])
+
+    def to_dict(self) -> TranslatableContentDict:
+        """Returns the dict representation of TranslatableContent object.
+
+        Returns:
+            TranslatableContentDict. A dict, mapping content_id, content_value
+            and content_type of a TranslatableContent instance to
+            corresponding keys 'content_id', 'content_value' and
+            'content_type'.
+        """
+        return {
+            'content_id': self.content_id,
+            'content_value': self.content_value,
+            'content_type': self.content_type
+        }
+
+
+class TranslatedContent:
+    """Class representing a translation of translatable content. For example,
+    if translatable content 'A' is translated into 'B' in a language other than
+    English, then 'B' is a TranslatedContent instance that represents this
+    class.
+    A (TranslatableContent) -----(translation)-----> B (TranslatedContent).
+
+    Args:
+        content_value: ContentValueType. Represents translation of translatable
+            content.
+        needs_update: bool. Whether any translation needs an update or not.
```

Comment:
I don't follow this. Isn't there just one translation here? If so, why "any translation"?

How come we don't need to store the language code here?

Reply:
Docstring updated.